Parser

NechushkinGit opened the pull request:
Added differentiation method (doesn't work with everything, still working on it)

Commit:
Adds parser template

luckynatrium committed:
polycop done, initializer done

luckynatrium committed:
add evaluate

NechushkinGit committed:
Merge pull request #1 from luckynatrium/polycop_feat

Polycop feat

NechushkinGit committed:
Added differentiation method

Wolwer1nE commented:
Fix code checks
Resolve merge confilcts

Wolwer1nE commented on lib/parser.rb:
```diff
@@ -0,0 +1,150 @@
+    end
+    def differentiate_inner(str) #string -> string
+      ###################################### --METHODS
+      def is_digit?(ch)
```
It will be better to define methods in class scope.

Vl-Tershch committed:
Changed and added new tests.

Commit:
merge master

Vl-Tershch committed:
Changed tests and configuration.

Vl-Tershch committed:
Merge branch 'master' into master

NechushkinGit committed:
Merge pull request #2 from DdiavaLL/master

Changed and added new tests.

Commit:
started new differentiation method

Commit:
Merge branch 'master' of https://github.com/NechushkinGit/silicium

luckynatrium committed:
add check for two or more variables in polynom. Move string transformation into separate method to_ruby_s

luckynatrium committed:
add check for two or more variables in polynom. Move string transformation into separate method to_ruby_s

luckynatrium committed:
Rename our module name,add support function to_kf, which will be used in finding real roots in polynom

NechushkinGit committed:
created new differentiation method

NechushkinGit committed:
deleted comments

NechushkinGit committed:
climate changes

luckynatrium committed:
add more supportive methods, encapsulated it in class PolynomRealRoots

luckynatrium committed:
add more supportive methods, encapsulated it in class PolynomRealRoots

NechushkinGit committed:
renamed file

luckynatrium committed:
after merge

luckynatrium committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	lib/parser.rb

NechushkinGit committed:
Merge pull request #4 from luckynatrium/master

PolynomRealRoots + Dimass stuff

Vl-Tershch committed:
Changed tests and configuration v2.

Vl-Tershch committed:
Changed tests and configuration v2.

Wolwer1nE commented:
Please, fix code climate issues and build issues.

kpdzhnv committed:
On branch master
	modified a few names and wrote most of8 the8 documentation

Changes to be committed:
	modified:   lib/algebra.rb

Vl-Tershch committed:
Changed tests and added new test.

NechushkinGit committed:
Merge pull request #6 from DdiavaLL/master

Update tests.

NechushkinGit committed:
Merge pull request #5 from kpdzhnv/master

On branch master

Dobrzhanskij committed:
Polynomial division class with own parser

NechushkinGit committed:
Merge pull request #7 from Dobrzhanskij/master

Polynomial division class with own parser

Dobrzhanskij committed:
Added precision to floating numbers

NechushkinGit committed:
Merge pull request #8 from Dobrzhanskij/master

Added precision to floating numbers

luckynatrium committed:
Move Differentiate to new file algebra_dif. Refactor step_up

luckynatrium committed:
Refactor hit_root

luckynatrium committed:
Do some shit

luckynatrium committed:
Merge master

NechushkinGit committed:
Merge pull request #9 from luckynatrium/master

Refactoring and climate changes

Vl-Tershch committed:
Changed project configuration.

NechushkinGit committed:
Merge pull request #10 from DdiavaLL/master

Changed project configuration.

Dobrzhanskij committed:
Methods' length corrected

Dobrzhanskij committed:
Delete polynomial_division.rb

Dobrzhanskij committed:
Rename polynom_division.rb to polynomial_division.rb

Dobrzhanskij committed:
Methods have been optimized

NechushkinGit committed:
Merge pull request #11 from Dobrzhanskij/master

Methods have been optimized

luckynatrium committed:
Refactor polycop and finish polynomrealroots

luckynatrium committed:
Refactor get_coef

luckynatrium committed:
Merge polynomRealRoots

NechushkinGit committed:
Merge pull request #12 from luckynatrium/master

PolynomRealRoots

Vl-Tershch committed:
Changed tests and added new for polynom division.

Dobrzhanskij committed:
Small bugs were fixed

NechushkinGit committed:
Merge pull request #13 from Dobrzhanskij/master

Small bugs were fixed

Dobrzhanskij committed:
Small fix

Vl-Tershch committed:
Small chenges.

Vl-Tershch committed:
Added tests for polynomial_division.

Wolwer1nE committed:
Merge pull request #54 from NechushkinGit/master

PolynomRootsReal

Wolwer1nE committed:
Makes polycop less complex

Wolwer1nE committed:
Makes rubocop happy with tests

Vl-Tershch committed:
Added new tests for polynomial_division.

NechushkinGit committed:
Merge pull request #15 from DdiavaLL/master

Architecture changes.

Dobrzhanskij committed:
Changes in division method 

Also now you can write without "1" or "-1" coefficient

NechushkinGit committed:
Merge branch 'master' into master

NechushkinGit committed:
Merge pull request #14 from Dobrzhanskij/master

Small fix

Dobrzhanskij committed:
Incorrect version

Dobrzhanskij committed:
Correct version

NechushkinGit committed:
Merge pull request #16 from Dobrzhanskij/master

Change wrong file into correct

Vl-Tershch committed:
Changed old tests for polynom division.

Vl-Tershch committed:
Changed old tests for polynom division v2.

NechushkinGit committed:
Merge pull request #17 from DdiavaLL/master

Changed old tests for polynom division.

Wolwer1nE committed:
Merge branch 'master' into refactoring

Wolwer1nE committed:
Fixes warnings in tests

Wolwer1nE committed:
Refactors polynom_parser

Wolwer1nE committed:
Merge branch 'master' into refactoring

Wolwer1nE committed:
Fixes some plotter problems

Wolwer1nE committed:
Fixes reduce interval return value

Wolwer1nE committed:
Merge branch 'refactoring' into master

Wolwer1nE committed:
Merge pull request #80 from NechushkinGit/master

Keeping parser at refactoring branch

Wolwer1nE committed:
Merge branch 'master' of https://github.com/mmcs-ruby/silicium into refactoring

Wolwer1nE committed:
Merge branch 'refactoring' of https://github.com/mmcs-ruby/silicium into refactoring

Wolwer1nE committed:
Fixes imports in algebra

Wolwer1nE committed:
Fixed Code Climate issues

NechushkinGit committed:
climate changes

NechushkinGit committed:
fixed an error

NechushkinGit committed:
fixed another error

NechushkinGit committed:
fixed ANOTHER error

Wolwer1nE committed:
Merge branch 'refactoring' into master

Wolwer1nE committed:
Merge pull request #85 from NechushkinGit/master

Updates Division module

Wolwer1nE committed:
Fixes polynomial division import (again)

NechushkinGit committed:
Merge pull request #19 from mmcs-ruby/refactoring

Refactoring
